### PR TITLE
Fix serial quirks for IXFR after restart

### PR DESCRIPTION
**Pull request by k0ekk0ek**
For details, see #227. This PR adds a test to verify the correct serial is used for IXFRs after a restart and a fix for the issue reported.

### CODE REVIEW

File: tpkg/ixfr_and_restart.tdir/ixfr_and_restart.dsc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+BaseName: ixfr_and_restart
+Version: 1.0
+Description: Verify NSD requests IXFR with correct serial after restart
+CreationDate: Thu Nov 01 14:11:27 CET 2020
```
Comment (wcawijngaards) [inline tpkg/ixfr_and_restart.tdir/ixfr_and_restart.dsc]:
```suggestion
CreationDate: Thu Nov 01 14:11:27 CET 2022
```
Typing error fixed. The date is not used to run the test framework, anyway.

File: tpkg/ixfr_and_restart.tdir/ixfr_and_restart.conf [hunk 2]
```diff
@@ -0,0 +1,24 @@
+server:
+  logfile: "nsd.log"
+  xfrdir: ""
+  xfrdfile: xfrd.state
```
Comment (wcawijngaards) [inline tpkg/ixfr_and_restart.tdir/ixfr_and_restart.conf]:
```suggestion
  xfrdfile: xfrd.state
  zonelistfile: "zone.list"
```
The test should specify a zonelistfile. The test does not use it, but it is neater to not refer to the system default zone list file from the unit test.

**Review by wcawijngaards — APPROVED** [review]
The changes look good.
It would be good to also fix xfrd.c:2359 compare_serial call parameters.

File: tpkg/ixfr_and_restart.tdir/ixfr_and_restart.dsc (hunk 1 above)
Comment (k0ekk0ek) [reply tpkg/ixfr_and_restart.tdir/ixfr_and_restart.dsc]: Included

File: tpkg/ixfr_and_restart.tdir/ixfr_and_restart.conf (hunk 2 above)
Comment (k0ekk0ek) [reply tpkg/ixfr_and_restart.tdir/ixfr_and_restart.conf]: Included

**Comment by k0ekk0ek** [comment]
> The changes look good. It would be good to also fix xfrd.c:2359 compare_serial call parameters.

Included this one too, good catch :+1:

**Comment by klaus-nicat** [comment]
Is this ready for testing?

**Comment by k0ekk0ek** [comment]
Hi @klaus-nicat, yes, it is.

**Review by wtoorop — APPROVED** [review]
Excellent! Thanks!

**Pull request merged**